src/Castle.Windsor/Core/Internal/Locking/Lock.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2004-2009 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Core.Internal.Locking
+{
+	public abstract class Lock
```

Comment:
I thought we were going to drop all interfaces and abstract classes, however this is a tested implementation so probably not a bad idea to just leave it.

Reply:
The interfaces are in use as return contracts. Thought better to keep.
The abstracts are also in use for substitution. Thought better to keep.